### PR TITLE
Feature/kubernetes

### PR DESCRIPTION
**Pull request by steeling**
add kubernetes files for the 2 new binaries

### CODE REVIEW

File: cmds/grpc-backend/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# From the root directory docker build -f cmds/grpc-backend/Dockerfile -t grpc-backend .
```
Comment (vosst) [inline cmds/grpc-backend/Dockerfile]: Nice start. I would propose a multi-stage docker build, though, reducing the final container size and just selecting the relevant executable:
```
FROM golang:latest AS build
ADD . /app
WORKDIR /app
RUN go mod download
RUN make

FROM ubuntu:latest
COPY --from=build /go/bin/grpc-backend /usr/bin
ENTRYPOINT ["/usr/bin/grpc-backend"]
```

File: cmds/http-gateway/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# From the root directory docker build -f cmds/http-gateway/Dockerfile -t http-gateway .
```
Comment (vosst) [inline cmds/http-gateway/Dockerfile]: See previous comment on multi-stage builds.

File: cmds/grpc-backend/main.go [hunk 3]
```diff
@@ -13,28 +14,31 @@ import (
...
 
-	l, err := net.Listen("tcp", address)
+	l, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
 	if err != nil {
 		return err
 	}
 	defer func() {
 		if err := l.Close(); err != nil {
-			logger.Error("Failed to close listener", zap.String("address", address), zap.Error(err))
+			logger.Error("Failed to close listener", zap.Int("port", port), zap.Error(err))
```
Comment (vosst) [inline cmds/grpc-backend/main.go]: Hmmm, I liked the `address` approach better as it allows for tying to a specific network interface.

File: config/make-files.py [hunk 4]
```diff
@@ -103,6 +101,11 @@ def join_str(self):
   with open(yaml_file, 'w') as f:
     check_call(['sed', 's/JOINLIST/%s/g;s/LOCALITYLIST/%s/g;s/PUBLIC_ADDR/%s/g' % (cr.join_str(), locality, cr.lb_ip), 'templates/cockroachdb-statefulset-secure.yaml'], stdout=f)
 
+for cr in create_clusters:
+  yaml_file = '%s/http-gateway.yaml' % (cr.directory)
+  with open(yaml_file, 'w') as f:
+    check_call(['sed', 's/YOUR_ZONE_HERE/%s/g' % (cr.zone), 'templates/http-gateway.yaml'], stdout=f)
+
```
Comment (jgrossmac) [inline config/make-files.py]: I'd love for us to move away from using these scripts to render variables in the charts and only use them for the purpose of generating certs and uploading them to the cluster.

File: pkg/dss/auth/auth.go [hunk 5]
```diff
@@ -16,16 +16,16 @@ type authClient struct {
...
-	bytes, err := ioutil.ReadFile(pkFile)
+	bytes, err := ioutil.ReadFile(f)
 	if err != nil {
 		return nil, err
 	}
 	return &authClient{publicKey: bytes}, nil
 }
 
 func (a *authClient) AuthInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+
```
[the remark this answers is not in the document]
Comment (steeling) [reply pkg/dss/auth/auth.go]: done!

File: config/make-files.py (hunk 4 above)
Comment (jgrossmac) [reply config/make-files.py]: @steeling once this is merged in, I can start building out proper helm charts and hopefully have a PR up by the end of the week.

File: config/templates/http-gateway.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+      - name: http-gateway
+        image: http-gateway:latest
+        imagePullPolicy: Never
+        ports:
+        - containerPort: 8080
+          name: http
+        command:
+          - "/bin/bash"
+          - "-ecx"
+          - "exec ./main.go  -grpc-backend=grpc-backend.YOUR_ZONE_NAME:TARGET_PORT -port=8080"
```
[the remark this answers is not in the document]
Comment (vosst) [reply config/templates/http-gateway.yaml]: +1 from my side for helm'ing.

File: config/make-files.py (hunk 4 above)
Comment (vosst) [reply config/make-files.py]: +1 for landing as is and moving forward while cleaning up the operational scripts.

**Pull request merged**